core/src/main/java/cz/o2/proxima/repository/EntityAwareAttributeDescriptor.java

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.repository;
+
+import com.google.common.base.Preconditions;
+import cz.o2.proxima.scheme.ValueSerializer;
+import cz.o2.proxima.storage.StreamElement;
+import java.net.URI;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+
+/** {@link AttributeDescriptor} */
+public class EntityAwareAttributeDescriptor<T> implements AttributeDescriptor<T> {
+
+  public static class Regular<T> extends EntityAwareAttributeDescriptor<T> {
+
+    private Regular(EntityDescriptor entity, AttributeDescriptor<T> attr) {
+      super(entity, attr);
+      Preconditions.checkArgument(!isWildcard());
+    }
+
+    /** Create upsert update to non-wildcard attribute. */
+    public StreamElement upsert(String uuid, String key, long stamp, T value) {
+      return StreamElement.upsert(
+          entity,
+          wrapped,
+          uuid,
+          key,
+          wrapped.getName(),
+          stamp,
+          wrapped.getValueSerializer().serialize(value));
+    }
+
+    public StreamElement upsert(String uuid, String key, Instant stamp, T value) {
```

Comment:
Big 👍 !
minor: Can you please add javadocs?

Reply:
Added.